tasks/sign-binaries/sign-binaries.yaml

```diff
@@ -0,0 +1,172 @@
+        echo -n "$windows_digest" > "$(results.unsignedWindowsDigest.path)"
+
+    # - name: sign-mac-binaries
+    #   image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+    #   script: |
+    #     #!/usr/bin/env bash
+    #     # TODO CLDX-79
+    - name: sign-windows-binaries
+      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      volumeMounts:
+        - name: windows-ssh-key-vol
+          mountPath: "/etc/secrets"
+          readOnly: true
+      env:
+        - name: WINDOWS_USER
+          valueFrom:
+            secretKeyRef:
+              name: $(params.windowsCredentials)
+              key: username
+        - name: WINDOWS_PORT
+          valueFrom:
+            secretKeyRef:
+              name: $(params.windowsCredentials)
+              key: port
+        - name: WINDOWS_HOST
+          valueFrom:
+            secretKeyRef:
+              name: $(params.windowsCredentials)
+              key: host
+        - name: QUAY_USER
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: username
+        - name: QUAY_PASS
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: password
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+
+        mkdir -p /root/.ssh
+        chmod 700 /root/.ssh
+        cp "/etc/secrets/id_rsa" /root/.ssh/id_rsa
+        cp "/etc/secrets/fingerprint" /root/.ssh/known_hosts
+        chmod 600 root/.ssh/known_hosts /root/.ssh/id_rsa
+
+        SSH_OPTS="-i /root/.ssh/id_rsa -o UserKnownHostsFile=/root/.ssh/known_hosts -p ${WINDOWS_PORT} \
+        ${WINDOWS_USER}@${WINDOWS_HOST}"
+        SCP_OPTS="-i /root/.ssh/id_rsa -o UserKnownHostsFile=/root/.ssh/known_hosts -P ${WINDOWS_PORT}"
+
+        unsigned_digest=$(cat "$(results.unsignedWindowsDigest.path)")
+        # Create the batch script
+        signing_script_file="/tmp/signing_script.bat"
+        set +x
+        cat << EOF > "$signing_script_file"
+
+        mkdir %TEMP%\$(params.pipelineRunUid) && cd /d %TEMP%\$(params.pipelineRunUid)
+        @echo off
+        oras login quay.io -u ${QUAY_USER} -p ${QUAY_PASS}
+        @echo on
+        oras pull $(params.quayURL)/unsigned@${unsigned_digest}
+
+        signtool sign /v /n "Red Hat" /fd SHA256 /tr http://timestamp.digicert.com /td SHA256 ^
+        %TEMP%\$(params.pipelineRunUid)\*
+
+        if errorlevel 1 (
+          echo Signing of binaries failed
+          exit /B %ERRORLEVEL%
+        )
+
+        signtool verify /v /pa %TEMP%\$(params.pipelineRunUid)\*
+
+        if errorlevel 1 (
+          echo Verification of binaries failed
+          exit /B %ERRORLEVEL%
+        )
+
+        echo [%DATE% %TIME%] Signing of Windows binaries completed successfully
+
+        oras push $(params.quayURL)/signed:$(params.pipelineRunUid) %TEMP%\$(params.pipelineRunUid) \
```

Comment:
I think you'd need the select-oci-auth piece before this, right @mmalina ? Like https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/create-pyxis-image/create-pyxis-image.yaml#L118

Reply:
I was doing `oras login` on line 119 with  a user:pass. 
Are you suggesting I use a config file with credentials instead? I'm not familiar with `select-oci-auth` and couldn't trace where it originates from that url.

Reply:
`select-oci-auth` originates from here: https://github.com/konflux-ci/oras-container/blob/main/hack/select-oci-auth.sh

If you're using a user+pass directly, then it doesn't apply. But it's a good question how we want to approach this. Because one way would be to rely on pull secrets linked to the service account - that's how all of our other pulls and pushes work - we don't do anything special (other than the `select-oci-auth` hack) and it just works, because those secrets are there. Hmm, but then you also need this working on the windows host. In theory you could set up a docker config there beforehand. But maybe it makes more sense to just use user+pass in this case.

(But note that I'm asking if we couldn't simply scp the files in another thread.)

Reply:
Thanks for sharing. I don't believe `select-oci-auth` applies here.

Instead of Quay image-pull-secrets we have a dedicated Quay org and robot/service account. The credentials are passed to Windows/mac hosts as part of the task during the SCP transfer.

Reply:
Yep, I agree, it should be fine.